packages/gatsby/cache-dir/develop-static-entry.js

```diff
@@ -79,6 +91,109 @@ export default (pagePath, callback) => {
     postBodyComponents = components
   }
 
+  const getPageDataPath = path => {
+    const fixedPagePath = path === `/` ? `index` : path
+    return join(`page-data`, fixedPagePath, `page-data.json`)
+  }
+
+  const getPageData = pagePath => {
+    const pageDataPath = getPageDataPath(pagePath)
+    const absolutePageDataPath = join(process.cwd(), `public`, pageDataPath)
+    const pageDataRaw = fs.readFileSync(absolutePageDataPath)
+
+    try {
+      return JSON.parse(pageDataRaw.toString())
+    } catch (err) {
+      return null
+    }
+  }
+
+  const pageData = getPageData(pagePath)
```

Comment:
This will include `staticQueryHashes`. We will want to map over those, read them from `public/page-data/sq/<hash>.json` and include that in the data that is passed in here to the Route Handler. 

Stuff after hydration will just work since the runtime is already aware of static queries and how to load them.